### DISCUSSION TITLE
page_stand_alone에서 폭포가 증가하는 문제

### DISCUSSION THREAD

**Discussion by devwontory** [KGU-CS-HOME/.github#13] [3. 이슈]
최근 프로젝트 규모가 커지면서 page_stand_alone을 통해 접근하는 페이지가 모두 먹통이 되는 이슈가 있었습니다.
현재는 일부 해결한 상태이나, 아직 해결해야 할 부분이 남아있습니다.

## 원인
기존의 page_stand_alone의 코드를 살펴보면 다음의 형태와 같습니다.
```
<c:choose>
 <c:when test="${boardType eq 'notice' }">
    <%@ include file="/WEB-INF/a.jsp"%>
  </c:when>
  <c:when test="${boardType eq 'guest' }">
    <%@ include file="/WEB-INF/b.jsp"%>
  </c:when>
  ...
  ... 반복
</c:choose>
```
이런 식의 코드를 작성했을 때는 분기 안의 모든 file이 컴파일됩니다.

### 왜 분기처리를 해주었는데도 불구하고 모든 파일이 컴파일될까?
그 이유는 분기는 jstl로 처리하지만, 안의 불러오는 문장은 jsp 문법이기 때문입니다.
코드는 jsp -> jstl -> html 순서로 처리됩니다.
그렇기 때문에 jsp 코드인 include 부분이 전부 처리되어왔던 것입니다.
프로젝트 규모가 커질수록 무리가 오는 것도 당연하겠죠.

&nbsp;

## 해결
이 문제는 jsp 코드인 include 부분을 jstl 구문으로 바꿔줌으로써 해결할 수 있었습니다.

제가 수정하고 난 이후의 코드 형태입니다.
```
<c:choose>
 <c:when test="${boardType eq 'notice' }">
    <c:import url="/WEB-INF/a.jsp"/>
  </c:when>
  <c:when test="${boardType eq 'guest' }">
    <c: import url="/WEB-INF/b.jsp"/>
  </c:when>
  ...
  ... 반복
</c:choose>
```
하지만 모든 페이지에서 jsp 코드를 jstl 구문으로 변경할 수는 없었습니다.
(bbs, register, professor를 제외하고는 모두 변경이 가능했습니다.)
위의 세 페이지를 jstl로 변경하였을 때는 페이지가 로딩되지 않는 문제가 있습니다.
> Failed to load resource: net::ERR_INCOMPLETE_CHUNKED_ENCODING

아직 원인을 찾지 못하기도 했고, 위의 세 페이지만 jsp 코드로 두어도 정상작동하는 것을 확인했기 때문에
일부만 변경해 둔 상태입니다.

다만, bbs, register, professor에서 작업이 계속 이루어지고, 규모가 커지면 언제 다시 생길지 모르는 오류입니다.
따라서 위의 세 페이지들도 jstl 구문으로의 변경이 필요하다고 생각됩니다.

**Comment by gabrielyoon7** [comment]
https://github.com/KGU-CS-HOME/webp/commit/f7161ed71bcee9e3d4b7fa83fde9ce4ecf6915c2

기존 학과 홈페이지에서도 비슷한 문제가 있어서 수정한 경험이 있는데 참고 해주시면 감사하겠습니다.

***수정 전***
```
<c:when test="${jsp == '\"information\"'}">
  <%@include file="/WEB-INF/jsp_v2/page/menu/information.jsp" %>
</c:when>
```

***수정 후***
```
<c:when test="${jsp == '\"information\"'}">
  <jsp:include page="/WEB-INF/jsp_v2/page/menu/information.jsp" flush="false"></jsp:include>
</c:when>
```

한번 시도해보시고 안되시면 다시 말씀해주세요~

**Reply by gabrielyoon7** [reply]
<%@include file="jsp주소" %>
에서

<jsp:include page="jsp주소" flush="false"></jsp:include>
형식으로 변경
